### ISSUE TITLE
binding_refresh_interval not having any impact

### ISSUE THREAD

**Issue by bapowell** [zauberzeug/nicegui#57]
Running on a Raspberry Pi, NiceGUI is fairly CPU intensive (although still usable). I tried changing the binding_refresh_interval value, via a ui.run() parameter, but it doesn't seem to make any difference on CPU usage.

**Comment by falkoschindler** [comment]
As far as I can tell, `binding_refresh_interval` does have an effect. Adding a print statement within the `loop()` in binding.py demonstrates the impact of that interval. I tried it with the following example:

```python3
from nicegui import ui

class Model:
    def __init__(self):
        self.x = 1

model = Model()
ui.number('x').bind_value_from(model, 'x')

ui.run(binding_refresh_interval=0.5)
```

There is probably something else CPU-intensive going on in your code. Can you boil it down to some minimum example?

**Comment by bapowell** [comment]
I think what I'm seeing is essentially the minimum CPU usage from NiceGUI. In other words, using your very minimal example, I see roughly 50% CPU usage on one of the RPi's cores. Changing the binding_refresh_interval has no impact.

What's interesting is that I can run JustPy, using the example at the end of https://justpy.io/tutorial/model_and_data/, which is also doing bind operations, and it runs with less than 5% CPU usage on one core.

Please understand this isn't a knock against NiceGUI, since it has a much more "batteries included" approach, which I can appreciate. But a 10x increase in CPU usage seems excessive, and I wonder if there's something I'm missing.

**Comment by rodja** [comment]
We will run our examples on an RPi in the next days and come back to you.

**Comment by me21** [comment]
I'm running my nicegui on the Beaglebone Black single board computer, it has one CPU core and it uses 20% give or take.
During startup, it loads CPU fully, but I can live with that.

**Comment by falkoschindler** [comment]
I just checked the CPU usage on a Raspberry Pi 3 Model B V1.2 (2015) with a simple hello-world script:

```python3
from nicegui import ui

ui.label('Hello world!')

ui.run()
```

Here are my conclusions:

- `reload=True` (the default) can be costly. Here is a related discussion over at uvicorn: https://github.com/encode/uvicorn/issues/338. If in doubt, try using `reload=False`.
- It might also help to limit the amount of files that need to be watched: Originally I tested with a test script in my home folder. This causes uvicorn to watch the whole home directory ramping the CPU up to 100 %. After moving the script into a separate folder, uvicorn had to watch only a single file, lowering the CPU to less than 1 %.
- `binding_refresh_interval` does not have any significant impact for this tiny script.

So if you don't need auto-reload, set it to `False`. Otherwise, keep the watched directory small (and/or use `reload_includes`, `reload_excludes` and `reload_dirs`).

Note that changes to the arguments of `ui.run` are not caught by the auto-reload but require restarting the script completely.
